python/3.12/Dockerfile

```diff
@@ -17,19 +18,20 @@ COPY *.patch /usr/src/
 RUN set -ex \
     && export PYTHON_VERSION=${PYTHON_VERSION} \
     && apk add --no-cache --virtual .fetch-deps \
-        gnupg \
         openssl \
         tar \
         xz \
+    && apk add --no-cache --virtual .cosign cosign \
+        --repository="https://dl-cdn.alpinelinux.org/alpine/v3.21/community" \
```

Comment:
The Python sigstore bundle use the new bundle format, the verification of which requires `cosign>=4.2.0`. Unfortunately alpine `3.21` is the earliest release with in this case version `4.2.1`. The additional repository is necessary until `3.19` and `3.20` are dropped.

There is a draft PR to add sigstore validation to the "official" Python docker images which uses a similar approach. https://github.com/docker-library/python/pull/983